template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/yarn/YarnResourceManagerRoleConfigProvider.java

```diff
@@ -20,13 +26,22 @@
 
     private static final String CAPACITY_SCHEDULER_CLASS = "yarn_resourcemanager_scheduler_class";
 
+    private static final String RESOURCE_MANAGER_YARN_SITE_SAFETY_VALVE = "resourcemanager_config_safety_valve";
+
     @Override
     protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
         switch (roleType) {
             case YarnRoles.RESOURCEMANAGER:
-                return List.of(
-                        config(CAPACITY_SCHEDULER_SAFETY_VALVE, ConfigUtils.getSafetyValveConfiguration(getCapacitySchedulerValveValue())),
-                        config(CAPACITY_SCHEDULER_CLASS, "org.apache.hadoop.yarn.server.resourcemanager.scheduler.capacity.CapacityScheduler"));
+                List<ApiClusterTemplateConfig> configs = new LinkedList<>();
+                if (isContainerPlacementConfigRequired(source)) {
+                    configs.add(config(RESOURCE_MANAGER_YARN_SITE_SAFETY_VALVE, getYarnSiteContainerPlacementSafetyValveValues()));
+                }
+                if (isLlapBasedConfigNeeded((CmTemplateProcessor) source.getBlueprintView().getProcessor())) {
```

Comment:
There is no LLAP in datahub clusters. It looks like there is a behavior change for pre-7.2.11 clusters with this change. Earlier we used to set this value always, but after this change, this will be only set if LLAP is enabled. A customer deleting and recreating the 7.2.9 cluster will see different values for the capacity scheduler.

Reply:
Yes. There's no LLAP. This code, in fact, should not exist. Don't want to make that change in this jira though.

In terms of a behaviour change - that is not the case, even though it looks like it.

Earlier isConfigurationNeeded would filter out setting the LLAP configs (that has the LLAP check). Now, since there are 2 different aspects which could trigger YARN RM configs, isConfigurationNeeded is more permissive, and whether the LLAP configs are required or not is checked again in this method.

Reply:
Thanks!